File: passwort_generator.cpp
#include <iostream>
#include <vector>
#include <fstream>
#include <ctime>
#include <chrono>

std::vector<std::string> passWord;
int passwordLength, wait;

void inputPasswordLength()
{
    std::cout << "Enter the lenght of your password: ";
    std::cin >> passwordLength;
}

int main()
{
    std::srand(std::time(0));
    inputPasswordLength();
    passWord.resize(passwordLength);

    // Start the timer
    auto start = std::chrono::high_resolution_clock::now();

    std::ofstream outputFile("passwords/passwordlist.txt");
    for (int i = 0; i < passwordLength; i++)
    {
        int random = 33 + std::rand() % 94;
        //std::cout << random << " ";
        char randomChar = random;

        passWord[i] = randomChar;
        //std::cout << randomChar << std::endl;
        // outputFile << random << " : ";
        outputFile << randomChar;
    }
    outputFile.close();

    // Stop the timer
    auto end = std::chrono::high_resolution_clock::now();

    // Calculate the duration
    std::chrono::duration<double> elapsed = end - start;
    std::cout << "Execution time: " << elapsed.count() << " seconds\n";

    std::cin >> wait;
    return 0;
}

File: passwort_generator.py
import random
import os
import time

def input_password_length():
    print("Waiting for input...")
    length = int(input("Enter the length of your password: "))
    print(f"Length received: {length}")
    return length

def generate_password(length):
    password = ""
    for _ in range(length):
        random_char = chr(random.randint(33, 126))
        password += random_char
    return password

def main():
    start_time = time.time()  # Start timing
    #print("Starting program...")
    password_length = input_password_length()
    password = generate_password(password_length)

    os.makedirs('passwords', exist_ok=True)
    with open('passwords/passwordlist-py.txt', 'w') as file:
        file.write(password)

    end_time = time.time()  # End timing
    #print(f"Password generated: {password}")
    print(f"Execution time: {end_time - start_time} seconds")
    wait = input("")
if __name__ == "__main__":
    main()
